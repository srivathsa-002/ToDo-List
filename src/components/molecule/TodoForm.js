import React from "react";
import shortid from "shortid";
import "../../styles/components/widget.css";
import InputBox from "../atoms/InputBox";
import AddTaskIcon from "../atoms/AddTaskIcon";

export default class TodoForm extends React.Component {
  state = {
    text: ""
  };

  handleChange = event => {
    this.setState({
      [event.target.name]: event.target.value
    });
  };

  handleSubmit = event => {
    event.preventDefault();
    this.props.onSubmit({
      id: shortid.generate(),
      text: this.state.text,
      complete: false
    });
    this.setState({
      text: ""
    });
  };

  render() {
    return (
      <form className="widget-header" onSubmit={this.handleSubmit} data-testid="form">
        <InputBox value={this.state.text} onChange={this.handleChange} />
        <div className="Grid-container" onClick={this.handleSubmit}>
          <AddTaskIcon />
          {/* <Grid container>
            <Grid item>
              <AddIcon />
              {"   "}
            </Grid>

            <Grid item style={{ paddingTop: "1px", paddingLeft: "5px" }}>
              <span style={{ paddingLeft: "5px" }}>Add task</span>
            </Grid>
          </Grid> */}
        </div>
      </form>
    );
  }
}
