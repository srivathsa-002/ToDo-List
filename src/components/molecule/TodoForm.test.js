import React from "react";
import {render,fireEvent} from "@testing-library/react";
import TodoForm from "./TodoForm";
import "@testing-library/jest-dom/extend-expect";
import {create} from "react-test-renderer";

describe("TodoForm testing", () => {
    test("Matches the snapshot", () => {
      const input = create(<TodoForm />);
      expect(input.toJSON()).toMatchSnapshot();
    });
});

it("renders correctly", () => {
    const{getByTestId} = render(<TodoForm />)

    expect(getByTestId("form")).toBeInTheDocument()
})