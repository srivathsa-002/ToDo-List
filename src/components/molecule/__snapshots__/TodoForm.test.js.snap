// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoForm testing Matches the snapshot 1`] = `
<form
  className="widget-header"
  data-testid="form"
  onSubmit={[Function]}
>
  <input
    className="widget-header-input"
    data-testid="input-field"
    name="text"
    onChange={[Function]}
    placeholder="Todo..."
    value=""
  />
  <div
    className="Grid-container"
    onClick={[Function]}
  >
    <div
      className="MuiGrid-root MuiGrid-container"
    >
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          data-testid="addTask"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
           
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
        style={
          Object {
            "paddingLeft": "5px",
            "paddingTop": "1px",
          }
        }
      >
        <span
          style={
            Object {
              "paddingLeft": "5px",
            }
          }
        >
          Add task
        </span>
      </div>
    </div>
  </div>
</form>
`;
